graylog2-web-interface/src/views/components/views/ViewList.jsx

```diff
@@ -1,126 +1,110 @@
 import { ButtonToolbar, DropdownButton, MenuItem } from 'components/graylog';
 import IfPermitted from 'components/common/IfPermitted';
 
 import { PaginatedList, SearchForm, Spinner, EntityList } from 'components/common';
 import View from './View';
 
-const ViewList = createReactClass({
-  propTypes: {
-    views: PropTypes.arrayOf(PropTypes.object),
-    pagination: PropTypes.shape({
-      total: PropTypes.number.isRequired,
-      page: PropTypes.number.isRequired,
-      perPage: PropTypes.number.isRequired,
-    }).isRequired,
-    handleSearch: PropTypes.func.isRequired,
-    handleViewDelete: PropTypes.func.isRequired,
-  },
-
-  getDefaultProps() {
-    return {
-      views: undefined,
-    };
-  },
-
-  getInitialState() {
-    return {
-      query: '',
-      page: 1,
-      perPage: 10,
-    };
-  },
-
-  componentDidMount() {
-    this.execSearch();
-  },
-
-  execSearch(resetLoadingState = () => {
-  }) {
-    const { query, page, perPage } = this.state;
-    this.props.handleSearch(query, page, perPage).then(resetLoadingState).catch(resetLoadingState);
-  },
-
-  handleSearch(query, resetLoadingState) {
-    this.setState({ query: query, page: 1 }, () => {
-      this.execSearch(resetLoadingState);
-    });
-  },
-
-  handleSearchReset() {
-    this.setState({ query: '', page: 1 }, () => {
-      this.execSearch();
-    });
-  },
-
-  handlePageChange(page, perPage) {
-    this.setState({ page: page, perPage: perPage }, () => {
-      this.execSearch();
-    });
-  },
-
-  handleViewDelete(view) {
+const itemActionsFactory = (view, onViewDelete) => {
+  return (
+    <IfPermitted permissions={['*']}>
+      <ButtonToolbar>
+        <DropdownButton title="Actions" id={`view-actions-dropdown-${view.id}`} bsSize="small" pullRight>
+          <MenuItem onSelect={onViewDelete(view)}>Delete</MenuItem>
+        </DropdownButton>
+      </ButtonToolbar>
+    </IfPermitted>
+  );
+};
+
+const ViewList = ({ pagination, handleSearch, handleViewDelete, views }) => {
+  const [query, setQuery] = useState('');
+  const [page, setPage] = useState(1);
+  const [perPage, setPerPage] = useState(10);
+
+  const execSearch = (resetLoadingState = () => {}) => {
+    handleSearch(query, page, perPage).then(resetLoadingState).catch(resetLoadingState);
+  };
+
+  useEffect(() => {
+    execSearch();
+  }, [query, page, setPage]);
+
+
+  const onSearch = (newQuery, resetLoadingState) => {
+    setQuery(newQuery);
+    setPage(1);
+    resetLoadingState();
+  };
+
+  const onSearchReset = () => {
+    setQuery('');
+    setPage(1);
+  };
+
+  const onPageChange = (newPage, newPerPage) => {
+    setPage(newPage);
+    setPerPage(newPerPage);
+  };
+
+  const onViewDelete = (view) => {
     return () => {
-      this.props.handleViewDelete(view).then(() => {
-        this.setState({ page: 1 }, () => {
-          this.execSearch();
-        });
+      handleViewDelete(view).then(() => {
+        setPage(1);
+        execSearch();
```

Comment:
We need to execute the search, when `page`, `perPage` or `query` changes, this works perfectly in combination with `useEffect`, except for this case (deleting a view). As you can see in the diff, we are running `execSearch ` after `setState`.  With the current implementation, `setPage(1)` will not trigger `executeSearch`, when the current page is 1.

The above ` execSearch();` in line 55 has no effect. also `setPage(1).then()` is not a possible solution.

Do you have any idea how to deal with this case?